File: C# Fundamentals/C# OOP Basics/11. InterfaceNAbstr/Ex/P01_IPerson/Citizen.cs
internal class Citizen : IPerson
{
    private string name;
    private int age;

    public string Name
    {
        get
        {
            return this.name;
        }
        private set
        {
            this.name = value;
        }
    }
    public int Age
    {
        get
        {
            return this.age;
        }
        private set
        {
            this.age = value;
        }
    }

    public Citizen(string name, int age)
    {
        this.Name = name;
        this.Age = age;
    }
}

File: Java/01.Advanced/02.OOP/09.Interfaces/Ex/src/t_04_food_shortage/Citizen.java
package t_04_food_shortage;

public class Citizen implements Person, Identifiable, Birthable, Buyer {
    private String name;
    private int age;
    private int food;
    private String id;
    private String birthDate;

    public Citizen(String name, int age, String id, String birthDate) {
        this.name = name;
        this.age = age;
        this.id = id;
        this.birthDate = birthDate;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public int getAge() {
        return age;
    }

    @Override
    public String getId() {
        return id;
    }

    @Override
    public String getBirthDate() {
        return birthDate;
    }

    @Override
    public String toString() {
        return name;
    }

    @Override
    public void buyFood() {
        food += 10;
    }

    @Override
    public int getFood() {
        return food;
    }
}
